File: cpp/image/downscale.cpp
#include <cv.h>
#include <highgui.h>

using namespace cv;
using namespace std;

int main(int argc, char *argv[])
{
  if (argc < 2){
    return -1;
  }
  Mat image = imread(argv[1]);
  Size size = (Size)image.size();
  Mat dst;
  pyrDown(image, dst, Size(size.width/2, size.height/2));
  namedWindow("in");
  namedWindow("out");
  imshow("in", image);
  imshow("out", dst);
  waitKey(10000); 
  destroyAllWindows();
  return 0;
}

File: python/image/downscale.py
# coding:utf-8

import cv2, sys
import numpy as np

if len(sys.argv) < 2:
    exit(-1)
img = cv2.imread(sys.argv[1], cv2.IMREAD_COLOR)
img_downscale = cv2.pyrDown(img)
cv2.namedWindow("before downscale", cv2.WINDOW_AUTOSIZE)
cv2.namedWindow("after downscale", cv2.WINDOW_AUTOSIZE)
cv2.imshow("before downscale", img)
cv2.imshow("after downscale", img_downscale)
cv2.waitKey(5000)
cv2.destroyAllWindows()
